### ISSUE TITLE
SCSS version

### ISSUE THREAD

**Issue by ghost** [modularscale/modularscale-sass#27]
could there be a .scss version of modular-scale.sass?

there's one, but it's outdated (canarymason/modular-scale)

**Comment by scottkellum** [comment]
No plans to but feel free to use `sass-convert` to generate your self a SCSS version. Keep in mind you will still need the Ruby code in your project to make it work with new versions.

Any particular reason why you want SCSS? Anything you are trying to modify?

**Comment by ghost** [comment]
thanks, I'm putting together a SCSS based toolkit, including modular-scale. 
It works great with compass and the gem installed, but including it directly doesn't work (think using SCSS without Compass), even after converting it to SCSS (with sass-convert).

as you asked, did you ever think about using 'ems' instead of pixels?
here's some code to do the conversion: https://gist.github.com/2237465
it works better for elastic layouts.

**Comment by scottkellum** [comment]
v1.0 requires Compass because a lot of the more complicated logic @canarymason ported to Ruby to speed things up. I highly recommend using this with Compass and adding a dependency like this: https://github.com/zurb/foundation/pull/773/files

If you need to have a straight Sass version you can dig back in the history to find one. Sass will work fine along side SCSS.

You can put any units through modular-scale. Just write something like `$base-size: 1em;` in your project. I have used it with em, rem, %, and px.

**Comment by ghost** [comment]
you're right, Scott
a scenario of using Sass without Compass may not be worth the time.

<code>$base-size: 1em</code> works fine,
but if using this function http://stackoverflow.com/questions/11926485/sass-function-to-convert-pixels-to-ems
like this <code>$base-size: pem(16px)</code> it says:
<code>Syntax error: Undefined operation: "11.08901 times 1em".</code>

is something wrong with my function?
thanks a lot for your time :)

**Comment by scottkellum** [comment]
Yeah, 

`$baseFontSize: 16;`

might work better with a unit and `!default` declaration

`$baseFontSize: 16px !default;`

Because Sass treats units like algebraic components they are added and canceled with these math functions. In order to remove units you can divide them like so `16px/16px = 1`. You can add a unit back by multiplying by `1em`. So using our result `1*1em` will give you your result.

I wrote a similar function here: https://github.com/scottkellum/Seasons/blob/master/sass/lib/tools/_relative-pixels.sass

**Comment by ghost** [comment]
great info and code

don't mind me saying but, the last function on there, shouln't it be named <code>rem($number, $em-offset)</code>?

i'm a Sass (amazed) newbie, everything's new :)

**Comment by scottkellum** [comment]
Good catch. thanks!

**Comment by jakejscott** [comment]
@sopppas Did you manage to convert this to scss?

**Comment by ghost** [comment]
@superlogical yes, here it is: http://footeste.info/_dip.scss

**Comment by snowlers** [comment]
@sopppas Did you ever convert the modular-scale.sass to an scss version? I need one since Web Workbench can't use sass version.

**Comment by scottkellum** [comment]
@madmarten pure SCSS or an SCSS version of the Compass plugin?

Note that modular-scale now requires Compass for performance reasons.

**Comment by snowlers** [comment]
Web Workbench (VS 2012) should be able to handle Compass, but they state that they don't support sass versions. And I can't get it to compile, the modular-scale throws errors. (I'm trying to get Zurb Foundation to work).

**Comment by jakejscott** [comment]
Hey @madmarten I was trying to do the exact same thing, get Web Workbench and Zurb Foundation working together! In the end though I just installed ruby and compass by following the instructions on this page

```
http://foundation.zurb.com/docs/gem-install.php
```

Honestly I haven't had any problems and its good to have to learn the syntax without any help from intellisense. I keep the compass reference open so I can learn that too:

```
http://compass-style.org/reference/compass/
```

The best part about using compass is you use the watch command, and it will detect when you make a change to any of your sass/scss files and then re-compile them.

```
compass install foundation
compass watch
```

I went one step further and added a SignalR based css file watcher, that live reloads the css in the page! So now I can view my web page, and make changes to my scss files, compass will compile any changes it detects and updates my css file. My SignalR file watcher detects this, and then notify's any connected clients to reload there stylesheets. Check it out here

```
https://gist.github.com/3895499
```

Let me know if wanna setup you project like this I can give you a hand!

Jake
